### ISSUE TITLE
[API Proposal]: Nullable.Create

### ISSUE THREAD

**Issue by mariusz96** [dotnet/runtime#105897]
### Background and motivation

The `Tuple` type has this nice `Create` method that frees you from specifying generic type parameters:
```csharp
public static class Tuple
{
    public static Tuple<T1, T2> Create<T1, T2>(T1 item1, T2 item2);
}
```
```csharp
var tuple = Tuple.Create(1, 2);
```

It would be nice if `Nullable` (there is already a static `Nullable` class) would have a similar method.

### API Proposal

```diff
namespace System;

public static class Nullable
{
+     public static Nullable<T> Create(T value) where T : struct
}
```

### API Usage

It would free you from specifying generic type parameters in code where you need to create/compare nullable and non-nullable structs:
```csharp
public enum MyEnum { FirstValue, SecondValue }
```
```csharp
IEnumerable<MyEnum> nullableEnums; // Already exist somehow.

var nonNullableEnums = nullableEnums.Cast<MyEnum?>();
```
```csharp
IEnumerable<MyEnum> nullableEnums; // Already exist somehow.

var nonNullableEnums = nullableEnums.Select(e => Nullable.Create(e));
```

**Comment by RenderMichael** [comment]
Notably, the provided example using `Nullable.Create` is more verbose than the `Cast` operation.

The 99% use case for things like this, in my experience, is casting a generic collection up to a wider type.

In .NET 9, `ReadOnlySpan<T>.CastUp` was introduced to do this exact thing, except on a single type. Maybe there could be more helpers in that vein? This could only be done by contra-variant/read-only collection, so the utility would be less narrower.

**Comment by huoyaoyuan** [comment]
> In .NET 9, `ReadOnlySpan<T>.CastUp` was introduced to do this exact thing, except on a single type. Maybe there could be more helpers in that vein

No. `CastUp` only works for same layout. The layout for nullable struct is different from underlying struct.

---

There are already cast syntax like `(MyStruct?)value`. When the target type is specified, the implicit cast can also be omitted. `Nullable.Create` is likely to be more verbose than the cast syntax.

**Comment by tannergooding** [comment]
This is unlikely to be provided. It is something that an interested developer could provide as their own extension method and otherwise is an anti-pattern that can trivially to degraded performance compared to the alternative. `Cast<T>` is preferred in such scenarios and avoids a per element callback.
